pyproject.toml

```diff
@@ -29,18 +29,63 @@ dependencies = [
     "tiktoken>=0.4.0",
 ]
 description = "LLM Chain for answering questions from docs"
-dynamic = ["optional-dependencies", "version"]
+dynamic = ["version"]
 keywords = ["question answering"]
 license = {file = "LICENSE"}
 maintainers = [
     {email = "jamesbraza@gmail.com", name = "James Braza"},
+    {email = "michael.skarlinski@gmail.com", name = "Michael Skarlinski"},
     {email = "white.d.andrew@gmail.com", name = "Andrew White"},
 ]
 name = "paper-qa"
 readme = "README.md"
 requires-python = ">=3.8"
 urls = {repository = "https://github.com/whitead/paper-qa"}
 
+[project.optional-dependencies]
+agents = [
+    "anyio",
+    "langchain-community",
+    "langchain-openai",
+    "pymupdf",
+    "requests",
+    "tantivy",
+    "tenacity",
+    "typer",
+    "typing_extensions",
+]
+dev = [
```

Comment:
Should we split out paperqa LLM dependencies (e.g. `anthropic`, `faiss-cpu`) into an `llm` extra?

Reply:
I'm not 100% what the best module split should be for these, we could split those out, but should we include langchain too? I need to test what needs to be in the import path to function.

Reply:
Previously we just had them all grouped into one section of `dev-requirements.txt`. We can just have one extra `llm` for now to keep it simple

Reply:
Yea I am just wondering if it makes sense to extricate langchain into the llm extra too.